drivers/sub8_rdi_dvl/include/rdi_explorer_dvl/driver.h

```diff
@@ -129,25 +129,38 @@ class Device {
           dirs.push_back(uf_common::make_xyz<geometry_msgs::Vector3>(0, +x, -z));
           dirs.push_back(uf_common::make_xyz<geometry_msgs::Vector3>(0, -x, -z));
         }
+        // Keep track of which beams didn't return for logging
+        std::vector<size_t> invalid_beams;
+        invalid_beams.reserve(4);
         for (int i = 0; i < 4; i++) {
           uf_common::VelocityMeasurement m;
           m.direction = dirs[i];
           int32_t vel = gets32le(ensemble.data() + offset + 2 + 4 * i);
           m.velocity = -vel * .01e-3;
           if (vel == -3276801) {  // -3276801 indicates no data
-            ROS_ERROR("DVL didn't return bottom velocity for beam %i", i + 1);
+            invliad_beams.push_back(i + 1);
```

Comment:
I can't compile the repo or test it (can't compile since the great merge broke my workspace and can't test since out of town), that's why I didn't PR it myself.